src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -129,7 +131,20 @@ public bool TryGetValue(object key, out object value, bool shouldCheckSystem)
 
 				return value;
 			}
-			set => Add(key, value);
+			set => Set(key, value);
+		}
+
+		private void Set(object key, object value)
```

Comment:
Would be good for `Add()` to call `Set()` as well to keep the `new LazyInitializer()` code together, adding a bool parameter to determine if it should throw if the key is already present. Not a big deal though. Otherwise looks good!